### PR TITLE
refactored identical ProteinRecName + ProteinAltName -> ProteinName.

### PR DESCRIPTION
**Pull request by eddturner**

### CODE REVIEW

File: core-domain/src/main/java/org/uniprot/core/uniprotkb/impl/UniProtKBEntryImpl.java [hunk 1]
```diff
@@ -562,7 +562,7 @@ private void updateProteinSectionEvidences(Set<Evidence> evidences, ProteinSecti
         }
     }
 
-    private void updateProteinRecNameEvidences(Set<Evidence> evidences, ProteinRecName pn) {
+    private void updateProteinRecNameEvidences(Set<Evidence> evidences, ProteinName pn) {
```
Comment (LeonardoGonzales) [inline core-domain/src/main/java/org/uniprot/core/uniprotkb/impl/UniProtKBEntryImpl.java:565]: I think updateProteinRecNameEvidences and updateProteinAltNameEvidences might be merged, right?

File: core-domain/src/test/java/org/uniprot/core/uniprotkb/description/impl/ProteinNameImplTest.java [hunk 2]
```diff
@@ -103,20 +99,70 @@ void testOnlyFullAndShort() {
...
                         new NameImpl(), Collections.emptyList(), Collections.emptyList());
-        ProteinRecName obj = ProteinRecNameBuilder.from(impl).build();
+        ProteinName obj = ProteinNameBuilder.from(impl).build();
 
         assertTrue(impl.hasFullName());
         assertTrue(impl.equals(obj) && obj.equals(impl));
         assertEquals(impl.hashCode(), obj.hashCode());
     }
+
+    public static ProteinName createProteinRecName(int listSize, boolean includeEvidences) {
```
Comment (LeonardoGonzales) [inline core-domain/src/test/java/org/uniprot/core/uniprotkb/description/impl/ProteinNameImplTest.java:119]: This method (createProteinRecName) and the below one (createProteinAltName) are identical, do we need to have both? Maybe a createProteinName would be enough.

Comment (LeonardoGonzales) [reply core-domain/src/test/java/org/uniprot/core/uniprotkb/description/impl/ProteinNameImplTest.java:119]: I think the other methods below can be optimised as well.. right?

File: tools-entry-scorer/src/main/java/org/uniprot/core/scorer/uniprotkb/ProteinDescriptionScored.java [hunk 3]
```diff
@@ -154,7 +137,7 @@ private void addECsFromSubName(List<ProteinSubName> proteinNames, Set<EC> ecs) {
                         .collect(Collectors.toSet()));
     }
 
-    private List<Evidence> extractProteinAltNameEvidences(ProteinAltName name) {
+    private List<Evidence> extractProteinAltNameEvidences(ProteinName name) {
```
Comment (LeonardoGonzales) [inline tools-entry-scorer/src/main/java/org/uniprot/core/scorer/uniprotkb/ProteinDescriptionScored.java:140]: I think that we might be able to merge this method with the Rec name equivalent.

**Review by LeonardoGonzales — COMMENTED** [review]
Just few details

File: core-domain/src/main/java/org/uniprot/core/uniprotkb/impl/UniProtKBEntryImpl.java (hunk 1 above)
Comment (eddturner) [reply core-domain/src/main/java/org/uniprot/core/uniprotkb/impl/UniProtKBEntryImpl.java:565]: ah good spot, done

File: tools-entry-scorer/src/main/java/org/uniprot/core/scorer/uniprotkb/ProteinDescriptionScored.java (hunk 3 above)
Comment (eddturner) [reply tools-entry-scorer/src/main/java/org/uniprot/core/scorer/uniprotkb/ProteinDescriptionScored.java:140]: done

File: core-domain/src/test/java/org/uniprot/core/uniprotkb/description/impl/ProteinNameImplTest.java (hunk 2 above)
Comment (eddturner) [reply core-domain/src/test/java/org/uniprot/core/uniprotkb/description/impl/ProteinNameImplTest.java]: done

**Pull request merged**